custom_endpoint/src/Controller/CustomEndpointController.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Drupal\custom_endpoint\Controller;
+
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Drupal\Core\Session\AccountProxy;
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Session\AccountInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Custom Endpoint Controller class
+ */
+class CustomEndpointController extends ControllerBase {
+
+  /**
+   * Custom Endpoint controller constructor
+   *
+   * @param \Drupal\Core\Session\AccountInterface
+   */
+  public function __construct(AccountInterface $currentUser) {
+    $this->currentUser = $currentUser;
```

Comment:
now we don't use a current user here so remove this
instead we want to create a class member (field) that will store the injected current_user_service